core/src/main/java/org/axonframework/messaging/StreamableMessageSource.java

```diff
@@ -35,4 +38,47 @@
      */
     MessageStream<M> openStream(TrackingToken trackingToken);
 
+    /**
+     * Creates the token at the beginning of an event stream.
+     *
+     * @return the token at the beginning of an event stream
+     */
+    default TrackingToken createTailToken() {
+        return null;
+    }
+
+    /**
+     * Creates the token at the end of an event stream.
```

Comment:
Perhaps it's good to be explicit about the fact that the "end" is where most recent events live?